controllers/space/space_controller_test.go

```diff
@@ -0,0 +1,796 @@
+
+					// then
+					require.NoError(t, err)
+					assert.Equal(t, reconcile.Result{Requeue: false}, res) // no need to explicitly requeue while the NSTemplate is terminating
+					nsTmplSet := nstemplatetsettest.AssertThatNSTemplateSet(t, test.MemberOperatorNs, "oddity", member1.Client).
+						Exists().
+						HasDeletionTimestamp().
+						HasConditions(nstemplatetsettest.Terminating()).
+						Get()
+					spacetest.AssertThatSpace(t, test.HostOperatorNs, "oddity", hostClient).
+						Exists().
+						HasStatusTargetCluster("member-1").
+						HasConditions(spacetest.Terminating()).
+						HasFinalizer(toolchainv1alpha1.FinalizerName) // finalizer is still present while the NSTemplateSet is not fully deleted
+
+					t.Run("requeue while NSTemplateSet is terminating", func(t *testing.T) {
+						// given another round of requeue without while NSTemplateSet is *not ready*
+						nsTmplSet.Status.Conditions = []toolchainv1alpha1.Condition{
+							{
+								Type:   toolchainv1alpha1.ConditionReady,
+								Status: corev1.ConditionFalse,
+								Reason: toolchainv1alpha1.NSTemplateSetTerminatingReason,
+							},
+						}
+						err := member1.Client.Update(context.TODO(), nsTmplSet)
+						require.NoError(t, err)
+						ctrl := newReconciler(hostClient, member1, member2)
+
+						// when
+						res, err := ctrl.Reconcile(context.TODO(), reconcile.Request{
+							NamespacedName: types.NamespacedName{
+								Namespace: space.Namespace,
+								Name:      space.Name,
+							},
+						})
+
+						// then
+						require.NoError(t, err)
+						assert.Equal(t, reconcile.Result{Requeue: false}, res) // no need to explicitly requeue while the NSTemplate is terminating
+						// no changes
+						nsTmplSet := nstemplatetsettest.AssertThatNSTemplateSet(t, test.MemberOperatorNs, "oddity", member1.Client).
+							Exists().
+							HasDeletionTimestamp().
+							HasConditions(nstemplatetsettest.Terminating()).
+							Get()
+						spacetest.AssertThatSpace(t, test.HostOperatorNs, "oddity", hostClient).
+							Exists().
+							HasStatusTargetCluster("member-1").
+							HasConditions(spacetest.Terminating()).
+							HasFinalizer(toolchainv1alpha1.FinalizerName)
+
+						t.Run("done when NSTemplateSet is deleted", func(t *testing.T) {
+							// given another round of requeue without with NSTemplateSet now *fully deleted*
+							member1Client.MockDelete = nil
+							err := member1.Client.Delete(context.TODO(), nsTmplSet)
+							require.NoError(t, err)
+							ctrl := newReconciler(hostClient, member1, member2)
+							// when
+							res, err := ctrl.Reconcile(context.TODO(), reconcile.Request{
+								NamespacedName: types.NamespacedName{
+									Namespace: space.Namespace,
+									Name:      space.Name,
+								},
+							})
+							// then
+							require.NoError(t, err)
+							assert.Equal(t, reconcile.Result{Requeue: false}, res) // no more requeue.
+							spacetest.AssertThatSpace(t, test.HostOperatorNs, "oddity", hostClient).
```

Comment:
you should also verify that NSTemplateSet is not there

Reply:
huh? it's already explicitly deleted in line 510:
```
member1Client.MockDelete = nil
err := member1.Client.Delete(context.TODO(), nsTmplSet)
```

Reply:
I meant that you should call the assertion to verify that the NSTemplateSet is still not there and the Space controller didn't create it.

Reply:
done in 5b4b684

Reply:
Thanks, but it looks like you added it to the other tests - not this particular one.

Reply:
yes, I did add it: https://github.com/codeready-toolchain/host-operator/pull/548/files#diff-92b9cf612f3442ae1d6b7359f59a2c5683ab7edb129c96fdc050bd4c083e5671R528-R529